src/nyx/features/ngldm.cpp

```diff
@@ -0,0 +1,346 @@
+#include <limits.h>
+#include "ngldm.h"
+#include "../environment.h"
+
+NGLDMfeature::NGLDMfeature() : FeatureMethod("NGLDMfeature")
+{
+	provide_features ({
+		NGLDM_LDE,
+		NGLDM_HDE,
+		NGLDM_LGLCE,
+		NGLDM_HGLCE,
+		NGLDM_LDLGLE,
+		NGLDM_LDHGLE,
+		NGLDM_HDLGLE,
+		NGLDM_HDHGLE,
+		NGLDM_GLNU,
+		NGLDM_GLNUN,
+		NGLDM_DCNU,
+		NGLDM_DCNUN,
+		NGLDM_GLM,
+		NGLDM_GLV,
+		NGLDM_DCM,
+		NGLDM_DCV,
+		NGLDM_DCE,
+		NGLDM_DCENE });
+}
+
+void NGLDMfeature::clear_buffers()
+{
+	f_LDE =
+		f_HDE =
+		f_LGLCE =
+		f_HGLCE =
+		f_LDLGLE =
+		f_LDHGLE =
+		f_HDLGLE =
+		f_HDHGLE =
+		f_GLNU =
+		f_GLNUN =
+		f_DCNU =
+		f_DCNUN =
+		f_GLCM =
+		f_GLV =
+		f_DCM =
+		f_DCV =
+		f_DCE =
+		f_DCENE = 0;
+}
+
+template <class PixelCloud> void NGLDMfeature::gather_unique_intensities (std::vector<PixIntens> & V, PixelCloud & C, PixIntens max_inten)

```

Comment:
should we use `const PixelCloud & C`?

Reply:
We should but we can't at the moment - a trivial pixel cloud argument would work as const while its oversized variant wouldn't. I need to make multiple changes in the latter that is a subject of a separate PR.